overlord/snapstate/snapmgr.go

```diff
@@ -46,27 +46,13 @@ type SnapManager struct {
 	runner *state.TaskRunner
 }
 
-// SnapSetupFlags are flags stored in SnapSetup to control snap manager tasks.
-type SnapSetupFlags Flags
-
-const (
-	// flags that are only used by SnapSetup grow downwards
-
-	// SnapSetupFlagRevert flags the SnapSetup as coming from a revert
-	SnapSetupFlagRevert SnapSetupFlags = 0x40000000 >> iota
-)
-
-func (f SnapSetupFlags) Revert() bool {
-	return f&SnapSetupFlagRevert != 0
-}
-
 // SnapSetup holds the necessary snap details to perform most snap manager tasks.
 type SnapSetup struct {
 	// FIXME: rename to RequestedChannel to convey the meaning better
 	Channel string `json:"channel,omitempty"`
 	UserID  int    `json:"user-id,omitempty"`
 
-	Flags SnapSetupFlags `json:"flags,omitempty"`
+	Flags Flags `json:"flags,omitempty"`
```

Comment:
Any reason not to have this as an embedded type, so all values are available directly as `snapst.DevMode`, etc?  This is a trivial change to do now because the code calling `Flags` explicitly still works, but we cannot do later because doing so changes the underlying JSON format.